File: app/src/main/java/com/adityaanand/morphdialogsample/MainActivity.java
package com.adityaanand.morphdialogsample;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.graphics.Color;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

import org.jetbrains.annotations.NotNull;

import java.util.Random;

import com.adityaanand.morphdialog.MorphDialog;
import com.adityaanand.morphdialog.interfaces.MorphSingleButtonCallback;
import com.adityaanand.morphdialog.utils.MorphDialogAction;
import com.adityaanand.morphdialogsample.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding ui;
    MorphDialog dialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ui = DataBindingUtil.setContentView(this, R.layout.activity_main);
    }

    public void morph(View view) {
        dialog = new MorphDialog.Builder(this, (FloatingActionButton) view)
                .title("Title")
                .content("This is a sentence. Here is another one.")
                .positiveText("Ok")
                .negativeText("Cancel")
                .neutralText("More")
                .onPositive(new MorphSingleButtonCallback() {
                    @Override
                    public void onClick(@NotNull MorphDialog dialog1, @NotNull MorphDialogAction which) {
                        Toast.makeText(MainActivity.this, "onPositive", Toast.LENGTH_SHORT).show();
                    }
                })
                .onNegative(new MorphSingleButtonCallback() {
                    @Override
                    public void onClick(@NotNull MorphDialog dialog1, @NotNull MorphDialogAction which) {
                        Toast.makeText(MainActivity.this, "onNegative", Toast.LENGTH_SHORT).show();
                    }
                })
                .onNeutral(new MorphSingleButtonCallback() {
                    @Override
                    public void onClick(@NotNull MorphDialog dialog1, @NotNull MorphDialogAction which) {
                        Toast.makeText(MainActivity.this, "onNeutral", Toast.LENGTH_SHORT).show();
                    }
                })
                .cancelable(false)
                .contentColor(Color.BLUE)
                .backgroundColor(Color.GREEN)
                .neutralColorRes(R.color.primary)
                .positiveColor(Color.BLACK)
                .titleColor(Color.YELLOW)
                .useDarkTheme(new Random().nextBoolean())
                .build();
        dialog.show();
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        MorphDialog.Companion.registerOnActivityResult(requestCode, resultCode, data)
                .forDialogs(dialog);
    }
}

File: app/src/main/java/com/adityaanand/morphdialogsample/MainActivity.kt
package com.adityaanand.morphdialogsample

import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast
import com.adityaanand.morphdialog.MorphDialog
import com.adityaanand.morphdialog.interfaces.MorphListCallbackMultiChoice
import com.adityaanand.morphdialog.interfaces.MorphSingleButtonCallback
import com.adityaanand.morphdialog.utils.MorphDialogAction
import com.adityaanand.morphdialogsample.databinding.ActivityMain2Binding
import java.util.*


class MainActivity : AppCompatActivity() {

    lateinit var ui: ActivityMain2Binding
    lateinit var dialog: MorphDialog

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ui = DataBindingUtil.setContentView(this, R.layout.activity_main2)
    }

    fun morph(view: View) {
        dialog = MorphDialog.Builder(this, view as FloatingActionButton)
                .title("Title")
                .iconRes(R.drawable.ic_launcher_background)
                .items("Item 1", "Item 2")
                // .content("This is a sentence. Here is another one.")
                .positiveText("Ok")
                .negativeText("Cancel")
                .neutralText("More")
                .itemsCallbackMultiChoice(object : MorphListCallbackMultiChoice {
                    override fun onSelection(dialog: MorphDialog, which: Array<Int>, texts: Array<CharSequence>) {
                        Toast.makeText(this@MainActivity, "onSelection${which.size}", Toast.LENGTH_SHORT).show()
                    }

                })
                .onPositive({ morphDialog: MorphDialog, morphDialogAction: MorphDialogAction ->
                    Toast.makeText(this@MainActivity, "onPositive", Toast.LENGTH_SHORT).show()
                })
                .onNegative(object : MorphSingleButtonCallback {
                    override fun onClick(a: MorphDialog, which: MorphDialogAction) {
                        Toast.makeText(this@MainActivity, "onNegative", Toast.LENGTH_SHORT).show()
                    }
                })
                .onNeutral(object : MorphSingleButtonCallback {
                    override fun onClick(dialog1: MorphDialog, which: MorphDialogAction) {
                        Toast.makeText(this@MainActivity, "onNeutral", Toast.LENGTH_SHORT).show()
                    }
                })
                //                .cancelable(false)
                .useDarkTheme(Random().nextBoolean())
                .build()
        dialog.show()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        MorphDialog.registerOnActivityResult(requestCode, resultCode, data)
                .forDialogs(dialog)
    }
}
